lib/code-ribbon-single-ribbon.js

```diff
@@ -50,11 +54,47 @@ class CodeRibbonSingleRibbon extends PaneAxis {
     };
   }
 
+  at_least_n_empty_right_patches(target_num = 2) {
+    if (target_num > this.children.length) return false;
+    for (var i = 1; i <= target_num; i++) {
+      if (
+        this.children[this.children.length -i].getItems().length > 0
```

Comment:
If we are in the single ribbon container, why are you dipping into the patch's children? I am not even sure what a patch's children would consist of for our sake besides normal editor stuff. Is this for checking if the patch has anything in it..?